packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/editModal.js

```diff
@@ -29,28 +25,43 @@ const EditModal = props => {
         handlePaymentError
     } = talonProps;
 
-    const paymentMethod =
-        selectedPaymentMethod === 'braintree' ? (
-            <div className={classes.body}>
-                <CreditCard
+    const paymentMethodComponent = useMemo(() => {
+        const isEditable = Object.keys(editablePayments).includes(
+            selectedPaymentMethod
+        );
+        if (isEditable) {
+            const PaymentMethodComponent =
+                editablePayments[selectedPaymentMethod];
+            return (
+                <PaymentMethodComponent
                     onDropinReady={handleDropinReady}
```

Comment:
Now that this is a general purpose component, perhaps we should change this prop to something more generic since they won't all have a "dropin".

```suggestion
                    onPaymentReady={handlePaymentReady}
```